executor/brie.go

```diff
@@ -265,37 +423,175 @@ func (b *executorBuilder) buildBRIE(s *ast.BRIEStmt, schema *expression.Schema)
+	e.info.storage = storageURL.String()
+
+	// handle s.Tables:
+	// IMPORT **s.Tables** FROM ...
+	switch {
+	case len(s.Tables) != 0:
+		tbls := make([]string, 0, len(s.Tables))
+		for _, tbl := range s.Tables {
+			if tbl.Schema.L == "" {
+				b.err = ErrBRIEInvalidTable.GenWithStackByArgs(fmt.Sprintf("please specify schema for %s in IMPORT", tbl.Name.O))
+				return nil
+			}
+			tbls = append(tbls, fmt.Sprintf("%s.%s", tbl.Schema, tbl.Name))
+		}
+		importTaskCfg.Mydumper.Filter = tbls
+	case len(s.Schemas) != 0:
+		dbs := make([]string, 0, len(s.Schemas))
+		for _, db := range s.Schemas {
+			dbs = append(dbs, fmt.Sprintf("%s.*", db))
+		}
+		importTaskCfg.Mydumper.Filter = dbs
+	default:
+		// no filter means pass all
+	}
+
+	// handle options
+	for _, opt := range s.Options {
+		switch opt.Tp {
+		case ast.BRIEOptionSkipSchemaFiles:
+			importTaskCfg.Mydumper.NoSchema = opt.UintValue != 0
+		case ast.BRIEOptionStrictFormat:
+			importTaskCfg.Mydumper.StrictFormat = opt.UintValue != 0
+		case ast.BRIEOptionCSVSeparator:
+			importTaskCfg.Mydumper.CSV.Separator = opt.StrValue
+		case ast.BRIEOptionCSVDelimiter:
+			importTaskCfg.Mydumper.CSV.Delimiter = opt.StrValue
+		case ast.BRIEOptionCSVHeader:
+			if opt.UintValue == ast.BRIECSVHeaderIsColumns {
+				importTaskCfg.Mydumper.CSV.Header = true
+			} else {
+				b.err = ErrBRIEInvalidOption.GenWithStackByArgs("CSV_HEADER", "only support FIELDS or COLUMNS to indicate it has header")
+				return nil
+			}
+		case ast.BRIEOptionCSVNotNull:
+			importTaskCfg.Mydumper.CSV.NotNull = opt.UintValue != 0
+		case ast.BRIEOptionCSVNull:
+			importTaskCfg.Mydumper.CSV.Null = opt.StrValue
+		case ast.BRIEOptionCSVBackslashEscape:
+			importTaskCfg.Mydumper.CSV.BackslashEscape = opt.UintValue != 0
+		case ast.BRIEOptionCSVTrimLastSeparators:
+			importTaskCfg.Mydumper.CSV.TrimLastSep = opt.UintValue != 0
+		case ast.BRIEOptionChecksum:
+			switch opt.UintValue {
+			case uint64(ast.BRIEOptionLevelOff):
+				importGlobalCfg.PostRestore.Checksum = importcfg.OpLevelOff
+			case uint64(ast.BRIEOptionLevelOptional):
+				importGlobalCfg.PostRestore.Checksum = importcfg.OpLevelOptional
+			default:
+				importGlobalCfg.PostRestore.Checksum = importcfg.OpLevelRequired
+			}
+		case ast.BRIEOptionAnalyze:
+			switch opt.UintValue {
+			case uint64(ast.BRIEOptionLevelOff):
+				importGlobalCfg.PostRestore.Analyze = importcfg.OpLevelOff
+			case uint64(ast.BRIEOptionLevelOptional):
+				importGlobalCfg.PostRestore.Analyze = importcfg.OpLevelOptional
+			default:
+				importGlobalCfg.PostRestore.Analyze = importcfg.OpLevelRequired
+			}
+		}
+	}
+
+	e.importTaskCfg = importTaskCfg
+	e.importGlobalCfg = importGlobalCfg
 
 	return e
 }
 
+func (b *executorBuilder) buildBRIE(s *ast.BRIEStmt, schema *expression.Schema) Executor {
+	tidbCfg := config.GetGlobalConfig()
+	if tidbCfg.Store != "tikv" {
+		b.err = ErrBRIERequireTiKV.GenWithStackByArgs(s.Kind, tidbCfg.Store)
```

Comment:
we should allow mocktikv for Lightning TiDB backend

Reply:
https://github.com/pingcap/tidb/pull/21195/commits/22ad9ce817ac310cd5371b1c1947819f05cbdcbb

just add a comment for this one, will switch to user specified lightning backend in next PRs